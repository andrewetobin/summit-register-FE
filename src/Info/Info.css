.Info {
  align-items: center;
  display: flex;
  flex-direction: column;
  background: url("../utilities/Images/mountains-2.jpg") no-repeat center
  rgba(165, 165, 165, 0.68);
  background-size: auto 100%;
  justify-content: space-evenly;
  overflow: auto;
}

.info-section {
  border: 1px solid white;
  border-radius: 20px;
  background: grey;
  opacity: .9;
  overflow: auto;
  padding: 20px;
}

.difficulty-icon {
  margin: 0 35%;
  width: 30%;
}

.view-log-btn {
  border: 1px solid black;
  border-radius: 20px;
  display: block;
  font-size: 16px;
  padding: 5px 10px;
}
