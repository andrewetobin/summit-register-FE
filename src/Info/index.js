import React, { Component } from "react";
import BackButton from "../BackButton";
import logo from "../utilities/images/logo.png";
import "./Info.css";

class Info extends Component {
  render() {
    const {
      name,
      altitude,
      difficulty,
      range
    } = this.props.currentMountainData.data.attributes;
    return (
      <div className="Info">
        <header className="header">
          <BackButton handleBackButton={this.props.handleBackButton} />
          <img className="logo-small" alt="logo" src={logo} />
        </header>
        <section className="info-section">
        <h1>{name}</h1>
        <h3>Altitude: {altitude} ft</h3>
        <h3>Difficulty: {difficulty}</h3>
        <h3>Range: {range}</h3>
        </section>
        <button className="view-log-btn" onClick={() => this.props.handleViewLogButton()}>View Log</button>
        <button className="sign-log-btn">Sign Log</button>
      </div>
    );
  }
}

export default Info;
