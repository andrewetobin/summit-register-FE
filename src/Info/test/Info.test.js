import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Info from "..";

describe("Info", () => {
  let wrapper;
  const currentLocation = {
    latitude: 39.6295333,
    longitude: -105.1153598,
    sum: -65.4858265
  };
  const currentMountainData = {
    id: "4",
    type: "mountain",
    attributes: {
      altitude: 14345,
      difficulty: "Black",
      name: "Blanca Peak",
      range: "Sangre de Cristo",
      summit: "37.577473,-105.485443",
      registries: {
        data: []
      }
    }
  };
  const currentMountainWeather = {
    summary: "Mostly cloudy throughout the day.",
    icon: "partly-cloudy-night",
    data: [
      {
        time: 1546041600,
        summary: "Mostly Cloudy",
        icon: "partly-cloudy-night",
        precipProbability: 0.55,
        temperature: 72.8,
        humidity: 0.95,
        windSpeed: 12.39
      }
    ]
  };
  beforeEach(() => {
    wrapper = shallow(
      <Info
        currentMountainData={currentMountainData}
        handleViewLogButton={jest.fn()}
        currentLocation={currentLocation}
        currentMountainWeather={currentMountainWeather}
      />
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Info
        currentMountainData={currentMountainData}
        handleViewLogButton={jest.fn()}
        currentLocation={currentLocation}
        currentMountainWeather={currentMountainWeather}
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  describe("toggleView", () => {
    it("display should toggle between 'info' and 'weather'", () => {
      wrapper.state().display = "info";
      wrapper.instance().toggleView();
      expect(wrapper.state().display).toEqual("weather");
    });

    it("display should toggle between 'info' and 'weather'", () => {
      wrapper.state().display = "weather";
      wrapper.instance().toggleView();
      expect(wrapper.state().display).toEqual("info");
    });
  });
});
