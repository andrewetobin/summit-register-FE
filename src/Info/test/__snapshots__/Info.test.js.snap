// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Info
    currentMountainData={
      Object {
        "altitude": 999999,
        "difficulty": "super hard",
        "name": "Mt Massive",
        "range": "Front Range",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="header"
        >
          <BackButton />
          <img
            alt="logo"
            className="logo-small"
            src="logo.png"
          />
        </header>,
        <h1>
          Mt Massive
        </h1>,
        <h3>
          Altitude: 
          999999
        </h3>,
        <h3>
          Difficulty: 
          super hard
        </h3>,
        <h3>
          Range: 
          Front Range
        </h3>,
        <button
          className="sign-log-btn"
        >
          Sign Log
        </button>,
      ],
      "className": "Info",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BackButton />,
            <img
              alt="logo"
              className="logo-small"
              src="logo.png"
            />,
          ],
          "className": "header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleBackButton": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "logo-small",
              "src": "logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mt Massive",
        },
        "ref": null,
        "rendered": "Mt Massive",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Altitude: ",
            999999,
          ],
        },
        "ref": null,
        "rendered": Array [
          "Altitude: ",
          999999,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Difficulty: ",
            "super hard",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Difficulty: ",
          "super hard",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Range: ",
            "Front Range",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Range: ",
          "Front Range",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Log",
          "className": "sign-log-btn",
        },
        "ref": null,
        "rendered": "Sign Log",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="header"
          >
            <BackButton />
            <img
              alt="logo"
              className="logo-small"
              src="logo.png"
            />
          </header>,
          <h1>
            Mt Massive
          </h1>,
          <h3>
            Altitude: 
            999999
          </h3>,
          <h3>
            Difficulty: 
            super hard
          </h3>,
          <h3>
            Range: 
            Front Range
          </h3>,
          <button
            className="sign-log-btn"
          >
            Sign Log
          </button>,
        ],
        "className": "Info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BackButton />,
              <img
                alt="logo"
                className="logo-small"
                src="logo.png"
              />,
            ],
            "className": "header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleBackButton": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "logo-small",
                "src": "logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mt Massive",
          },
          "ref": null,
          "rendered": "Mt Massive",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Altitude: ",
              999999,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Altitude: ",
            999999,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Difficulty: ",
              "super hard",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Difficulty: ",
            "super hard",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Range: ",
              "Front Range",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Range: ",
            "Front Range",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Log",
            "className": "sign-log-btn",
          },
          "ref": null,
          "rendered": "Sign Log",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
